src/nunit.analyzers/ClassicModelAssertUsage/AreEqualClassicModelAssertUsageCodeFix.cs

```diff
@@ -45,16 +50,18 @@ protected override void UpdateArguments(Diagnostic diagnostic, List<ArgumentSynt
                         SyntaxFactory.SingletonSeparatedList(toleranceArgumentNoColon)));
             }
 
-            arguments.Insert(2, SyntaxFactory.Argument(equalToInvocationNode));
-
-            // Then we have to remove the 1st argument because that's now in the "Is.EqualTo()"
-            arguments.RemoveAt(0);
-
-            // ... and if the 3rd argument was a double, that has to go too.
-            if (hasToleranceValue)
+            var arguments = new List<ArgumentSyntax>() { actualArgument, SyntaxFactory.Argument(equalToInvocationNode) };
+            var handledParameterNames = new[]
             {
-                arguments.RemoveAt(2);
-            }
+                NUnitFrameworkConstants.NameOfExpectedParameter,
+                NUnitFrameworkConstants.NameOfActualParameter,
+                NameOfDeltaParameter,
+            };
+            return arguments
+                .Concat(argumentNamesToArguments
```

Comment:
`arguments.AddRange` would be better.
However as the only remaining parameters for the _message_ would need the same treatment for all CodeFixes it is better dealt with in the caller.

Reply:
@manfred-brands I think there was also `format`. Could you point me to the code that handles `message` and `format`?

Reply:
I'm also curious why `AddRange` is better.

Reply:
Duh, found the answer to my first question:
```cs
CodeFixHelper.UpdateStringFormatToFormattableString(arguments, this.MinimumNumberOfParameters);
```

Reply:
> I'm also curious why `AddRange` is better.

For one it is simpler. 
Second it doesn't have to copy the existing members (even though there are only two).
Third it doesn't allocate a complete new list but only extends the existing one (which might not require any allocations)